src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -440,5 +444,21 @@ private static void GetLocationInfo(Document document, Location location, out Te
             originalLineInfo = location.GetLineSpan();
             mappedLineInfo = location.GetMappedLineSpan();
         }
+
+        /// <summary>
+        /// Properties for a diagnostic generated by an explicit build.
+        /// </summary>
+        internal static ImmutableDictionary<string, string> PropertiesForBuildDiagnostic => s_Properties;
```

Comment:
Just moved it to a common place as now there are multiple clients that want to know if a diagnostic came from build.